parallel-task-set/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::sync::Arc;
+use tokio::sync::Semaphore;
+use tokio::task::JoinSet;
+
+const DEFAULT_MAX_PARALLELISM: usize = 16;
```

Comment:
nit, take it or leave it: I might make this a `pub` associated constant on `ParallelTaskSet`